anyway/parsers/infographics_data_cache_updater.py

```diff
@@ -13,6 +13,50 @@
 db = SQLAlchemy(app)
 
 
+def is_cache_eligible(news_flash):
+    return (
+        news_flash.accident
+        and news_flash.resolution in (["כביש בינעירוני"])
+        and news_flash.road_segment_name is not None
+    )
+
+
+def is_in_cache(nf):
+    return (
+        len(CONST.INFOGRAPHICS_CACHE_YEARS_AGO)
+        == db.session.query(InfographicsDataCache)
+        .filter(InfographicsDataCache.news_flash_id == nf.get_id())
+        .count()
+    )
+
+
+def add_news_flash_to_cache(news_flash):
+    if not is_cache_eligible(news_flash):
+        logging.debug(
+            f"add_news_flash_to_cache: news flash does not qualify:{news_flash.serialize()}"
+        )
+        return
+    try:
+        db.get_engine().execute(
+            InfographicsDataCache.__table__.insert(),  # pylint: disable=no-member
+            [
+                {
+                    "news_flash_id": news_flash.get_id(),
+                    "years_ago": y,
+                    "data": anyway.infographics_utils.create_infographics_data(
+                        news_flash.get_id(), y
+                    ),
+                }
+                for y in CONST.INFOGRAPHICS_CACHE_YEARS_AGO
+            ],
+        )
+    except Exception as e:
+        logging.error(
+            f"Exception while inserting to cache. flash_id:{news_flash})"
+            f":cause:{e.__cause__}, class:{e.__class__}"
+        )
```

Comment:
[`logging.exception()`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) does the exception logging for you.